### PR TITLE
Fix ts erros for ErrorPage

### PR DESCRIPTION
**Pull request by TomNUSDS**
- Fix ts errors and tighten up types in general.
- Add unit tests ErrorPage.

### CODE REVIEW

File: frontend-react/src/pages/details/Details.tsx [hunk 1]
```diff
@@ -26,7 +25,7 @@ function useQuery() {
 
 const DetailsContent = () => {
     let queryMap = useQuery();
-    let reportId = queryMap["reportId"];
+    let reportId = queryMap ? ["reportId"] : "";
```
Comment (kevinhaube) [inline frontend-react/src/pages/details/Details.tsx:28]: Wait, does this work the same way as `queryMap ? queryMap["reportId"] : "" `??

**Review by kevinhaube — APPROVED** [review]
Thank you for doing this! Love how clean things look with TS :)

File: frontend-react/src/pages/details/Details.tsx (hunk 1 above)
Comment (kevinhaube) [reply frontend-react/src/pages/details/Details.tsx:28]: I also believe I've used optional chaining like `queryMap?.["reportId"] : ""`. I love finding new little shortcuts like these!

Comment (TomNUSDS) [reply frontend-react/src/pages/details/Details.tsx:28]: Good catch! Yeah, this seems like a mistake. 
Seems like: `const reportId = queryMap?.["reportId"] || "";`  would make more sense.

File: frontend-react/src/pages/error/ErrorPage.tsx [hunk 2]
```diff
@@ -1,50 +1,52 @@
...
    e.g. <ErrorPage code={CODES.NOT_FOUND_404} />
  */
 export enum CODES {
     UNSUPPORTED_BROWSER = "unsupported-browser",
     NOT_FOUND_404 = "not-found",
+    UNKNOWN = "unknown-error",
+}
+
+interface ErrorPageProps {
+    code?: CODES.UNSUPPORTED_BROWSER | CODES.NOT_FOUND_404 | CODES.UNKNOWN;
```
Comment (TomNUSDS) [inline frontend-react/src/pages/error/ErrorPage.tsx:19]: @kevinhaube This limits the possible values to a subset of values from CODES... but I wonder if I shouldn't just use the enum as a type?  `code?: CODES` ? (LOL.. code reviewing my own stuff.)

File: frontend-react/src/pages/error/ErrorPage.tsx [hunk 3]
```diff
@@ -1,50 +1,52 @@
...
+    error?: string;
+    errorInfo?: React.ErrorInfo;
+    type?: "page" | "message";
 }
 
-function ErrorPageWrapper({ children }: JSX.Element) {
+const ErrorPageWrapper = (props: React.PropsWithChildren<ErrorPageProps>) => {
```
Comment (TomNUSDS) [inline frontend-react/src/pages/error/ErrorPage.tsx:25]: `React.PropsWithChildren` is a nice way to do do this. The `props` then has `props.children` automatically but also can contain OTHER props, too.

File: frontend-react/src/pages/details/Details.tsx (hunk 1 above)
Comment (TomNUSDS) [reply frontend-react/src/pages/details/Details.tsx]: Fixed.

File: frontend-react/src/pages/error/ErrorPage.tsx (hunk 2 above)
Comment (TomNUSDS) [reply frontend-react/src/pages/error/ErrorPage.tsx]: Fixed

Comment (kevinhaube) [reply frontend-react/src/pages/error/ErrorPage.tsx]: I saw this and thought "man, it sucks we'd have to update both this and the enum. I had no idea you could use an enum as a type, though! My brain thought about whether we should make a type, but ultimately we would've just been changing where we updated the list of values from here to that type. The enum typing is great!

**Pull request merged**